Add new RFC for new docs revamp (#84)
giantswarm/rfc: platform-docs-migration/README.md

---
creation_date: 2024-02-01

> issues: [https://github.com/giantswarm/giantswarm/issues/29503]

Author: ```suggestion
issues:
- https://github.com/giantswarm/giantswarm/issues/29503
```

---
creation_date: 2024-02-01
issues:
- https://github.com/giantswarm/giantswarm/issues/29503
owners:
- https://github.com/orgs/giantswarm/teams/team-horizon
state: review

> summary: Revamp our docs to be Platform Dev ready considering new Cluster API (CAPI) architecture. Instead of modifying existing doc content find a way to create those from a clean plate and avoid stale content.

Author: ```suggestion
summary: Revamp our docs to be Platform Dev ready considering the new Cluster API (CAPI) architecture. Instead of modifying existing docs content find a way to create a new site from a clean plate and avoid stale content.
```

---

> # Revamp our docs
> 
> ## Intro
> 
> As part of the Documentation Overhaul that Horizon is working on during first quarter of 2024 there was idea of how to transition for our current docs to a desired state where the new Dev Platform product is described correctly and precisely.
> 
> ## State of the documentation
> 
> Our [current documentation hub](https://docs.giantswarm.io) is an ongoing effort to maintain old Vintage entries that not longer are valid for our new product and at the same time adequate to the new Dev Platform concept. Those are the current works streams our docs have today:
> 
> 1) Horizon and SIG docs maintaining a coherent docs structure that follow new product developments.
> 2) Teams updating doc entries for out of date documents. Most of the times they don`t have the time to rewrite the content to make it Platform/CAPI ready, so they stays out of date.
> 3) Teams, KaaS and platform, adding new content that is only working for the new implementations (CAPI).
> 
> ## Goals
> 
> - To have a good documentation portal that does not have entries out of date. Only new implementation (since we have completely deprecated Vintage).
> - Stop trying to bring current articles to life, most of them are not longer valid in new implementation.
> - To have a period for transition. Customer or support still have access to Vintage docs but new docs are also available.
> 

Author: ```suggestion
# Revamp our docs

## Intro

As part of the [Documentation Overhaul](https://github.com/giantswarm/giantswarm/issues/29503) that Horizon is working on during the first quarter of 2024 there was an idea of how to transition our current docs to a desired state where the new Dev Platform product is described correctly and precisely.

## State of the documentation

Our [current documentation hub](https://docs.giantswarm.io) is an ongoing effort to maintain old Vintage entries that are no longer valid in our new product, and at the same time adequate to the new Dev Platform and CAPI concept. 

Today we have several work streams trying to achieve that goal:

1) Horizon and SIG docs maintaining a coherent docs structure that follows new product developments.
2) Teams updating doc entries for out-of-date documents. Most of the time they don`t have the time to rewrite the content to make it Platform/CAPI ready, so they stay out of date.
3) Teams, KaaS and platform, adding new content that is only working for the new implementations (CAPI).

## Goals

The described scenario is not working and ideally we find an alternative that has these goals:

- Have a good documentation portal that does not have entries out of date. Only new implementation (since we have completely deprecated Vintage).
- Stop trying to bring current articles to life, most of them are no longer valid in the new implementation. Just add what is relevant and possibly identify spots to fill by teams.
- Have a period for transition where everyone can look for content in Vintage and Dev Platform docs. Customers or support still have access to Vintage docs but new docs are also available.

```

---
creation_date: 2024-02-01
issues:
- https://github.com/giantswarm/giantswarm/issues/29503
owners:
- https://github.com/orgs/giantswarm/teams/team-horizon
state: review

> summary: Revamp our docs to be Platform Dev ready considering the new Cluster API (CAPI) architecture. Instead of modifying existing docs content find a way to create a new site from a clean plate and avoid stale content.

Reviewer: This is a very vague summary without actions. Let's put the main points in here.

```suggestion
summary: Revamp our docs to describe the Dev Platform product, considering the new Cluster API (CAPI) architecture. Temporarily create a second domain and create the new content from scratch instead of modifying existing docs. Start with content skeleton and also copy some existing content that can go directly into the new site.
```

Today we have several work streams trying to achieve that goal:

1) Horizon and SIG docs maintaining a coherent docs structure that follows new product developments.
2) Teams updating doc entries for out-of-date documents. Most of the time they don`t have the time to rewrite the content to make it Platform/CAPI ready, so they stay out of date.
3) Teams, KaaS and platform, adding new content that is only working for the new implementations (CAPI).

## Goals

The described scenario is not working and ideally we find an alternative that has these goals:

- Have a good documentation portal that does not have entries out of date. Only new implementation (since we have completely deprecated Vintage).
- Stop trying to bring current articles to life, most of them are no longer valid in the new implementation. Just add what is relevant and possibly identify spots to fill by teams.
- Have a period for transition where everyone can look for content in Vintage and Dev Platform docs. Customers or support still have access to Vintage docs but new docs are also available.

> ## Process
> 
> Instead of creating a Pull Request in [our docs repository](https://github.com/giantswarm/docs) where we work in parallel (accumulating lot of changes and drifts in configuration), we create a new clean repository for the new docs. This new repo can be deployed in a temporal domain where we can visualize the changes till we reach a point of satisfaction and customers are migrated from Vintage product.
> 
> **But why?**
> 
> - In the new repository we can creating a structure according to the new necessities without having to transition from existing content.
> - We don`t need to maintain aliases or links to old entries or sections.
> - It is way easier to organize the work between teams to add needed minimal content for every section.
> - We can work in iterations.
> - Both docs hub will be publicly available.
> - The end result of docs will be simpler and cleaner since we are not influence by old structure and content.
> 
> __Note__: Personally I was trying to deprecate a doc entry last week and spent more than an hour because link dependencies and inconsistencies in the content.
> 
> **How?**
> 
> General steps to achieve the goals
> 
> 1) Create new repo for new docs hub. Just the skeleton, configuration, CI/CD and new domain/server.
> 2) Create the main structure. Already discussions [here](https://miro.com/app/board/uXjVO2Dh15w=/).
> 3) Migrate the part of the content already exist in the current docs.
> 4) Create tickets for the teams with missing documentation pages.
> 5) Once migrated all customer (or almost), move new docs to become our current docs. We can still maintain old docs in new domain (vintage-docs.giantswarm.io).

Author: ```suggestion
## Process

Instead of creating a huge Pull Request in [our docs repository](https://github.com/giantswarm/docs) where we work in parallel (accumulating a lot of changes and configuration drifts), I propose to create a new clean repository for the new docs. This new repo can be deployed in a temporal (sub)domain where we can visualize the changes till we reach a point of satisfaction and furthermore, customers are migrated from Vintage product.

**But why?**

- In the new repository we can create a structure according to the new necessities without having to transition from existing content.
- We don`t need to maintain aliases or links to old entries or sections.
- It is way easier to organize the work between teams to add needed minimal content for every section.
- We can work in iterations.
- Both docs sites will be publicly available.
- The end result of our docs will be simpler and cleaner since we are not influenced by old structure and content.

__Note__: Personally I was trying to deprecate a doc entry last week and spent more than an hour because of linked dependencies and inconsistencies in the content.

**How?**

General steps to achieve the goals

1) Create a new repo for the new docs hub. Just the skeleton, configuration, CI/CD and new domain/server.
2) Create the main structure. Already discussions [here](https://miro.com/app/board/uXjVO2Dh15w=/).
3) Migrate the part of the content that already exists in the current docs. Adapt to the new structure and style.
4) Create tickets for the teams with missing documentation pages.
5) Once we migrate all customers (or almost), move new docs to become our current docs. We can still maintain old docs in the a (sub)domain (vintage-docs.giantswarm.io).
```

> General steps to achieve the goals
> 
> 1) Create a new repo for the new docs hub. Just the skeleton, configuration, CI/CD and new domain/server.
> 2) Create the main structure. Already discussions [here](https://miro.com/app/board/uXjVO2Dh15w=/).
> 3) Migrate the part of the content that already exists in the current docs. Adapt to the new structure and style.
> 4) Create tickets for the teams with missing documentation pages.
> 5) Once we migrate all customers (or almost), move new docs to become our current docs. We can still maintain old docs in the a (sub)domain (vintage-docs.giantswarm.io).
> 

Author: ```suggestion
General steps to achieve the aforementioned goals_

1) Create a new repo for the new docs hub. Just the skeleton, configuration, CI/CD and new domain/server.
2) Work iteratively by top-level section (Ex: Platform Overview, Getting Started,...)
  2.1) Create the top-level structure proposal. It will be mapped from [here](https://miro.com/app/board/uXjVO2Dh15w=/). When SIG docs and teams approve it, create the structure empty.
  2.2) Identity owners for the sections to be completed or created or adapted. Create tickets for teams. Help and guide them to complete it. Think of the docs entry structure to have a similar outline for similar pages.
3) Iterate over step 2 till all top-level sections are done
5) Once we migrate all customers (or almost), move new docs to become our current docs. We can still maintain old docs in the a (sub)domain (vintage-docs.giantswarm.io).

```

---
creation_date: 2024-02-01
issues:
- https://github.com/giantswarm/giantswarm/issues/29503
owners:
- https://github.com/orgs/giantswarm/teams/team-horizon
state: review
summary: Revamp our docs to describe the Dev Platform product, considering the new Cluster API (CAPI) architecture. Temporarily create a second domain and create the new content from scratch instead of modifying existing docs. Start with content skeleton and also copy some existing content that can go directly into the new site.
---

# Revamp our docs

## Intro

As part of the [Documentation Overhaul](https://github.com/giantswarm/giantswarm/issues/29503) that Horizon is working on during the first quarter of 2024 there was an idea of how to transition our current docs to a desired state where the new Dev Platform product is described correctly and precisely.

## State of the documentation

> Our [current documentation hub](https://docs.giantswarm.io) is an ongoing effort to maintain old Vintage entries that are no longer valid in our new product, and at the same time adequate to the new Dev Platform and CAPI concept. 

Author: ```suggestion
Our [current documentation hub](https://docs.giantswarm.io) is an ongoing effort to maintain old Vintage entries that are no longer valid in our new product, and at the same time adequate to the new Dev Platform and CAPI concept.
```

- Stop trying to bring current articles to life, most of them are no longer valid in the new implementation. Just add what is relevant and possibly identify spots to fill by teams.
- Have a period for transition where everyone can look for content in Vintage and Dev Platform docs. Customers or support still have access to Vintage docs but new docs are also available.

## Process

Instead of creating a huge Pull Request in [our docs repository](https://github.com/giantswarm/docs) where we work in parallel (accumulating a lot of changes and configuration drifts), I propose to create a new clean repository for the new docs. This new repo can be deployed in a temporal (sub)domain where we can visualize the changes till we reach a point of satisfaction and furthermore, customers are migrated from Vintage product.

**But why?**

- In the new repository we can create a structure according to the new necessities without having to transition from existing content.
- We don`t need to maintain aliases or links to old entries or sections.
- It is way easier to organize the work between teams to add needed minimal content for every section.
- We can work in iterations.

> - Both docs sites will be publicly available.

Reviewer: Maybe we can consider deployment to the same domain, but with a clear selection between 2 "contents". This way, all URLs can remain intact and we still start from scratch, rendering content from a different directory for CAPI stuff.

[Example from competitor that offers a version selection dropdown](https://docs.d2iq.com/dkp/2.7/)

This needs not block the RFC, but maybe we can write this down as technical alternative to shortly try.

- The end result of our docs will be simpler and cleaner since we are not influenced by old structure and content.

__Note__: Personally I was trying to deprecate a doc entry last week and spent more than an hour because of linked dependencies and inconsistencies in the content.

**How?**

General steps to achieve the aforementioned goals_

1) Create a new repo for the new docs hub. Just the skeleton, configuration, CI/CD and new domain/server.
2) Work iteratively by top-level section (Ex: Platform Overview, Getting Started,...)

>   2.1) Create the top-level structure proposal. It will be mapped from [here](https://miro.com/app/board/uXjVO2Dh15w=/). When SIG docs and teams approve it, create the structure empty.

Reviewer: _Who_ is going to collaborate to create the structure? I'm wondering how we can fix the abovementioned "Today we have several work streams". Should we be specific here?

>   2.2) Identity owners for the sections to be completed or created or adapted. Create tickets for teams. Help and guide them to complete it. Think of the docs entry structure to have a similar outline for similar pages.

Reviewer: `s/Identity/Identify/`?

3) Iterate over step 2 till all top-level sections are done
5) Once we migrate all customers (or almost), move new docs to become our current docs. We can still maintain old docs in the a (sub)domain (vintage-docs.giantswarm.io).

## Few clarifications

- This is a proposal, feedback welcome.
- I have consider migration within the repo and it will be way complex and difficult.

> - We don`t plan to change the style at the moment.

Reviewer 2: I know this is out of scope for this RFC, but I was thinking if this might be a chance for us to align docs with internal docs and handbook, i.e. using docsy across all 3, or do we think that would be a lot of work? Main reasons would be single tool, but also getting more "upstream" features ootb.

---

> creation_date: 2024-02-07

Author: ```suggestion
creation_date: 2024-02-19
```

issues:
- https://github.com/giantswarm/giantswarm/issues/29503
owners:
- https://github.com/orgs/giantswarm/teams/team-horizon

> state: review

Author: ```suggestion
state: approved
```

**But why?**

- In the new path we can create a structure according to the new necessities without having to transition from existing content.
- We don`t need to maintain aliases or links to old entries or sections.
- It is way easier to organize the work between teams to add needed minimal content for every section.
- We can work in iterations.
- Both docs sites will be publicly available.
- The end result of our docs will be simpler and cleaner since we are not influenced by old structure and content.

**How?**

General steps to achieve the aforementioned goals:

1) Moved the `vintage` content to a subpath so we leave a clean space in the top level docs site.
2) Add aliases from old content entries to include the old path so we don`t break links.
3) Leave the main docs entry page where it is now but pointing to the vintage path, so the appearance still the same.

> 4) Avoid bots to scrape the docs for now on to avoid temporal indexes.

Author: ```suggestion
4) Avoid bots to scrape the docs from now on to avoid temporal indexes. Add a banner to the vintage path to warn users of the state.
```

5) Work iteratively by top-level section adding new content (Ex: Platform Overview, Getting Started,...)
  2.1) Team Horizon will create a the top-level structure proposal. It will be mapped from [here](https://miro.com/app/board/uXjVO2Dh15w=/). When SIG docs and teams approve it,  we will create the structure empty.
  2.2) Identify owners for the sections to be completed, created or adapted. Create tickets for teams. Help and guide them to complete it. Think of the docs entry structure to have a similar outline for similar pages.
6) Iterate over step 2 till all top-level sections are done and we are happy with the current state. At this point change the main docs entry to point back to the top level docs and add a banner to inform this is our current documentation and the old one is under vintage subpath.
7) Once we migrate all customers (or almost) from vintage, we can decide to remove the vintage folder.

## Few clarifications

- This is a proposal, feedback welcome.

> - I have consider migration within the repo and it will be way complex and difficult.

Author: ```suggestion
- I have considered the migration will take time and this approach will allow teams to take their time till we decide to make it public.
```
